src/main/java/ca/corefacility/bioinformatics/irida/repositories/sample/MetadataEntryRepositoryImpl.java

```diff
@@ -37,50 +37,53 @@ public MetadataEntryRepositoryImpl(DataSource dataSource, EntityManager entityMa
 		parameters.addValue("project", project.getId());
 
 		/*
-		 * The below code for loading the metadata fields could be simplified into a single query, but the previous
-		 * 3-join query was taking 10s of seconds to load for large projects.  The 2 query process below for loading the
-		 * fields is providing significantly better performance than loading fields in a single 3-join query.
+		 * The below code for loading the metadata fields could be simplified
+		 * into a single query, but the previous 3-join query was taking 10s of
+		 * seconds to load for large projects. The 2 query process below for
+		 * loading the fields is providing significantly better performance than
+		 * loading fields in a single 3-join query.
 		 */
 
-		//First get all the field IDs that are associated with this project
+		// First get all the field IDs that are associated with this project
 		String fieldIdQueryString = "SELECT DISTINCT e.field_id FROM metadata_entry e INNER JOIN project_sample p ON e.sample_id=p.sample_id WHERE p.project_id=:project";
 		parameters.addValue("project", project.getId());
 		List<Long> fieldIds = tmpl.queryForList(fieldIdQueryString, parameters, Long.class);
 
 		List<MetadataTemplateField> fields;
 		if (!fieldIds.isEmpty()) {
-			//next load the full fields
-			String queryString = "SELECT * from metadata_field f WHERE f.id IN :fields";
-			Query nativeQuery = entityManager.createNativeQuery(queryString, MetadataTemplateField.class);
+			// next load the full fields
+			String queryString = "SELECT f from MetadataTemplateField f WHERE f.id IN :fields";
+			TypedQuery<MetadataTemplateField> nativeQuery = entityManager.createQuery(queryString,
```

Comment:
Switched to `TypeQuery` from `Query` which is type safe and removed unsafe type conversion warning.